File: src/main/java/org/mitallast/queue/crdt/log/ReplicatedLog.java
package org.mitallast.queue.crdt.log;

import javaslang.collection.Vector;
import org.mitallast.queue.common.codec.Message;

import java.io.Closeable;

public interface ReplicatedLog extends Closeable {

    long index();

    LogEntry append(long id, Message event);

    Vector<LogEntry> entriesFrom(long index);

    void delete();

    @Override
    void close();
}

File: src/main/java/org/mitallast/queue/crdt/log/ReplicatedLog.kt
package org.mitallast.queue.crdt.log

import javaslang.collection.Vector
import org.mitallast.queue.common.codec.Message

import java.io.Closeable

interface ReplicatedLog : Closeable {

    fun index(): Long

    fun append(id: Long, event: Message): LogEntry

    fun entriesFrom(index: Long): Vector<LogEntry>

    fun delete()

    override fun close()
}
